frontend/src/components/paletteTestLeo/PaletteTestLeo.jsx

```diff
@@ -0,0 +1,309 @@
+import axios from "axios";
+import { useState } from "react";
+import { AiFillLock } from "react-icons/ai";
+import "./PaletteTestLeo.css";
+
+export default function PaletteTestLeo({ labelAndColorArray }) {
+  // palette top//
+  const labels = labelAndColorArray.map((elt) => elt[0]);
+  const currentColors = labelAndColorArray.map((elet) => elet[1]);
+  const setters = labelAndColorArray.map((elt) => elt[2]);
+
+  // palette bottom//
+  const [fetchType, setFetchType] = useState("complementary");
+  const [fetchedColors, setFetchedColors] = useState([]);
+  const [lockedColors, setLockedColors] = useState([]);
+  const [hoveredDiv, setHoveredDiv] = useState(null);
+  const [colorPerDivItem, setColorsPerDivItem] = useState([
+    null,
+    null,
+    null,
+    null,
+    null,
+  ]);
+
+  const handleChangeColor = (i) => {
+    const nextColor =
+      colorPerDivItem[i] === null || colorPerDivItem[i] === 4
+        ? fetchedColors[0]
+        : fetchedColors[colorPerDivItem[i] + 1];
+    setters[i](nextColor);
```

Comment:
Interdit de mettre à jours des props, attention, pour mettre à jours des valeurs dériver des props il faut faire un state

Reply:
je te laisse regarder ces sources là si tu crois que c'est une mauvaise pratique : https://stackoverflow.com/questions/65284652/is-it-a-correct-approach-to-pass-usestate-setter-to-a-child-for-updating-parents et https://stackoverflow.com/questions/35537229/how-can-i-update-the-parents-state-in-react